compute/baremetal.py

```diff
@@ -0,0 +1,139 @@
+"""Support VM lifecycle operation in an OpenStack Cloud."""
+import logging
+from typing import List, Optional
+
+from libcloud.compute.base import Node
+
+LOG = logging.getLogger()
+
+# libcloud does not have a timeout enabled for Openstack calls to
+# ``create_node``, and it uses the default timeout value from socket which is
+# ``None`` (meaning: it will wait forever). This setting will set the default
+# to a magical number, which is 280 (4 minutes). This is 1 minute less than the
+# timeouts for production settings that should allow enough time to handle the
+# exception and return a response
+
+
+# Custom exception objects
+class ResourceNotFound(Exception):
+    pass
+
+
+class ExactMatchFailed(Exception):
+    pass
+
+
+class VolumeOpFailure(Exception):
+    pass
+
+
+class NetworkOpFailure(Exception):
+    pass
+
+
+class NodeError(Exception):
+    pass
+
+
+class NodeDeleteFailure(Exception):
+    pass
+
+
+class CephBaremetalNode:
+    """Represent the VMNode required for cephci."""
+
+    def __init__(
+        self,
+        **params,
+    ) -> None:
+        """
+        Initialize the instance using the provided information.
+
+        The co
+
+        Args:
+            username:   The name of the user to be set for the session.
```

Comment:
We need a mechanism to pass credentials. Not all environments would support password based access. Is there a prerequisite of user account?